File: jenjin-core/src/main/java/com/binarymonks/jj/layers/Layer.java
package com.binarymonks.jj.layers;


import com.badlogic.gdx.InputMultiplexer;

public abstract class Layer {

    protected InputMultiplexer inputMultiplexer = new InputMultiplexer();
    protected LayerStack stack;

    public abstract void update();

    public void setStack(LayerStack layerStack) {
        stack=layerStack;
    }

    public void removeSelf(){
        if(stack!=null){
            stack.remove(this);
        }
    }
}

File: jenjin-core/src/main/kotlin/com/binarymonks/jj/core/layers/Layer.kt
package com.binarymonks.jj.core.layers


import com.badlogic.gdx.InputMultiplexer

abstract class Layer {

    var inputMultiplexer = InputMultiplexer()
    var stack: LayerStack? = null

    abstract fun update()

    fun removeSelf() {
        if (stack != null) {
            stack!!.remove(this)
        }
    }
}
